dynawo/sources/Models/Modelica/Dynawo/Electrical/Transformers/NetworkTransformer.mo

```diff
@@ -0,0 +1,117 @@
+within Dynawo.Electrical.Transformers;
+
+/*
+* Copyright (c) 2024, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite
+* of simulation tools for power systems.
+*/
+
+model NetworkTransformer "Two windings transformer with a fixed ratio, same model as the Network one."
+
+/* Equivalent circuit and conventions:
+
+               I1  r                I2
+    U1,P1,Q1 -->---oo------R+jX-----<-- U2,P2,Q2
+  (terminal1)          |               (terminal2)
+                      G+jB
+                       |
+                      ---
+*/
+  extends Dynawo.Electrical.Controls.Basics.SwitchOff.SwitchOffTransformer;
+  extends AdditionalIcons.Transformer;
+
+  Dynawo.Connectors.ACPower terminal1(V(re(start = u10Pu.re), im(start = u10Pu.im)), i(re(start = i10Pu.re), im(start = i10Pu.im))) annotation(
+    Placement(visible = true, transformation(origin = {-100, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {-100, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Dynawo.Connectors.ACPower terminal2(V(re(start = u20Pu.re), im(start = u20Pu.im)), i(re(start = i20Pu.re), im(start = i20Pu.im))) annotation(
+    Placement(visible = true, transformation(origin = {100, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {100, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+
+  Types.ActivePowerPu P1Pu "Active power on side 1 in pu (base SnRef) (receptor convention)";
+  Types.ReactivePowerPu Q1Pu "Reactive power on side 1 in pu (base SnRef) (receptor convention)";
+  Types.ActivePowerPu P2Pu "Active power on side 2 in pu (base SnRef) (receptor convention)";
+  Types.ReactivePowerPu Q2Pu "Reactive power on side 2 in pu (base SnRef) (receptor convention)";
+
+  Types.VoltageModulePu U1Pu "Voltage on side 1 in pu (base U1Nom)";
+  Types.VoltageModulePu U2Pu "Voltage on side 2 in pu (base U2Nom)";
+
+  Types.CurrentModulePu I1Pu "Current on side 1 in pu (base U1Nom, SnRef) (receptor convention)";
+  Types.CurrentModulePu I2Pu "Current on side 2 in pu (base U2Nom, SnRef) (receptor convention)";
+
+  Real ISide1 "Current on side 1 in A (receptor convention)";
+  Real ISide2 "Current on side 2 in A (receptor convention)";
```

Comment:
You could put : Types.CurrentModule as a type ... in kA.
And remove the 1000 in  factorPuToASide1 = **1000**. * SystemBase.SnRef / (sqrt(3.) * U1Nom); and factorPuToASide2

Reply:
In the equivalent Network model we have the same 1000 factor so I wanted to have something to be equivalent. I would keep them both or remove them both

Reply:
I would say it's better to avoid having large numbers by removing the 1000 factor.

Reply:
Also no need to keep **Real** when adding **Types.CurrentModule**

Reply:
I think type is ok now

Reply:
For the factor I would keep it in this pr and see if we can change it in the network model as well as it is something we can plot in the platform side or might use